libraries/networking/src/FingerprintUtils.cpp

```diff
@@ -143,18 +113,52 @@ QString FingerprintUtils::getMachineFingerprintString() {
 
                 if (RegQueryValueEx(cryptoKey, MACHINE_GUID_KEY, NULL, NULL,
                                     reinterpret_cast<LPBYTE>(&machineGUID[0]), &guidSize) == ERROR_SUCCESS) {
-                    uuidString = QString::fromStdString(machineGUID);
+                    hash.addData(QString::fromStdString(machineGUID).toUtf8());
+                    success = true;
                 }
             }
         }
 
         RegCloseKey(cryptoKey);
     }
 
+    if (!success) {
+        // Let getMachineFingerprint handle this contingency
+        return QUuid().toString();
+    }
 #endif //Q_OS_WIN
 
-    return uuidString;
+    // Makes this hash unique to us
+    hash.addData("Vircadia");
+
+    // Stretching
+    for (int i=0; i < 65535; i++) {
+        hash.addData(hash.result());
+    }
+
+    QByteArray result = hash.result();
+    result.resize(128 / 8); // GUIDs are 128 bit numbers
+
+    // Set UUID version to 4, ensuring it's a valid UUID
+    result[6] = (result[6] & 0x0F) | 0x40;
 
+    // Of course, Qt couldn't be nice and just parse something like:
+    // 1b1b9d6d45c2473bac13dc3011ff58d6
+    //
+    // So we have to turn it into:
+    // {1b1b9d6d-45c2-473b-ac13-dc3011ff58d6}
+
+    QString uuidString = result.toHex();
+    uuidString.insert(20, '-');
+    uuidString.insert(16, '-');
+    uuidString.insert(12, '-');
+    uuidString.insert(8, '-');
+    uuidString.prepend("{");
+    uuidString.append("}");
+
+    qCDebug(networking) << "Final machine fingerprint:" << uuidString;
```

Comment:
Should this be logged?
Review other qCDebug() and qCWarning() statements in this file, too?

Reply:
The original code logged the data, yes